tests/basic/ini_parse_quantity_basic.phpt

```diff
@@ -0,0 +1,29 @@
+--TEST--
+ini_parse_quantity() function - basic test for ini_parse_quantity()
+--FILE--
+<?php
+foreach (array(
+    '-1',
+    '0',
+    '1',
+    '1b',
+    '1k',
+    '1m',
+    '1g',
+    '1gb',
+    '14.2mb',
+    'boat'
+) as $input) {
+    echo ini_parse_quantity( $input ) . PHP_EOL;
+}
+?>
+--EXPECT--
+0
+0
+1
+1
+1024
+1048576
+1073741824
+14
+0
```

Comment:
This test looks good, but it'll fail. 😄 

To run tests locally, use the `run-tests.php` script in the root of php-src. After you build PHP with your changes (using `make`), you can run the test like this:

```shell
php run-tests.php -p sapi/cli/php tests/basic/ini_parse_quantity_basic.phpt
```

Where `-p sapi/cli/php` is pointing to the new PHP executable that you just built that includes your changes.

There are many options you can pass to `run-tests.php`. You can check them out by entering `php run-tests.php -h`.

When you have a failing test, `run-tests.php` will create a bunch of additional files in the same directory alongside your test file, which you can use for debugging.

For example, when `tests/basic/ini_parse_quantity_basic.php` fails, you'll see the following new files created:

```
tests/basic/ini_parse_quantity_basic.diff
tests/basic/ini_parse_quantity_basic.expm
tests/basic/ini_parse_quantity_basic.log
tests/basic/ini_parse_quantity_basic.out
tests/basic/ini_parse_quantity_basic.php
tests/basic/ini_parse_quantity_basic.sh
```

Take a look at them to see what all they contain. `*.out` will tell you what the text actually printed out, while `*.diff` will show a diff of what was expected vs. what was actually printed.

For more information, check out the following:

* https://qa.php.net/write-test.php
* https://qa.php.net/phpt_details.php
* https://qa.php.net/expectf_details.php

Reply:
This is all great, and I'm quite happily surprised at how easy the testing setup is. Nothing wrong with test that fail before they succeed either 😄 

These run locally for me, but I couldn't build them when I rebased against `master` (I think because of changes introduced in 8.1? Could not build PHP with the following error on `make`

```
make: *** No rule to make target `/Users/dmsnell/code/php-src/main/php_stdint.h', needed by `ext/opcache/ZendAccelerator.lo'.  Stop.
```

I'll try rebasing again in a work day or two as I presume this will be fixed upstream quickly (it's there for me in `master` anyway).

Maybe I'll hold off on a more exhaustive test suite until the interplay with #7951 is clearer. The `zend_test_ini_parse_quantity` functions there I think could be renamed as `ini_parse_quantity` and replace this entire patch 🙃